edb/pgsql/compiler/dml.py

```diff
@@ -372,8 +372,12 @@ def fini_dml_stmt(
     # referencing this class yields the expected results.
     dml_stack = get_dml_stmt_stack(ir_stmt, ctx=ctx)
     if isinstance(ir_stmt, irast.InsertStmt):
+        # The union CTE might have a SELECT from an ELSE clause, which
+        # we don't actually want to include.
+        assert len(parts.dml_ctes) == 1
+        cte = next(iter(parts.dml_ctes.values()))[0]
         relctx.add_type_rel_overlay(
-            ir_stmt.subject.typeref, 'union', union_cte,
+            ir_stmt.subject.typeref, 'unIon', cte,
```

Comment:
```suggestion
            ir_stmt.subject.typeref, 'union', cte,
```

Reply:
Not sure about the assertion here.  How would this work on updates to parent types?

Reply:
You most likely need to reassemble the `UNION` here without the `ELSE` part.

Reply:
Oops on the `unIon`! I added that to make sure this is where a particular union in the generated code was coming from :P

Reply:
This branch is only for INSERTs. The assert should always be true.